OpenDreamRuntime/Procs/Native/DreamProcNativeRoot.cs

```diff
@@ -2122,6 +2122,84 @@ public static DreamValue NativeProc_sorttext(DreamObject instance, DreamObject u
             return new DreamValue(soundObject);
         }
 
+        [DreamProc("splicetext")]
+        [DreamProcParameter("Text", Type = DreamValueType.String)]
+        [DreamProcParameter("Start", Type = DreamValueType.Float, DefaultValue = 1)]
+        [DreamProcParameter("End", Type = DreamValueType.Float, DefaultValue = 0)]
+        [DreamProcParameter("Insert", Type = DreamValueType.String, DefaultValue = "")]
+        public static DreamValue NativeProc_splicetext(DreamObject instance, DreamObject usr, DreamProcArguments arguments)
+        {
+            arguments.GetArgument(0, "Text").TryGetValueAsString(out var text);            
+            arguments.GetArgument(1, "Start").TryGetValueAsInteger(out var start);
+            arguments.GetArgument(2, "End").TryGetValueAsInteger(out var end);             
+            arguments.GetArgument(3, "Insert").TryGetValueAsString(out var insert_text);
+
+            if(text == null) 
+                if(insert_text == null | insert_text == "")
+                    return DreamValue.Null;
+                else
+                    return new DreamValue(insert_text);
+            else if(text == "")
+                return new DreamValue(insert_text);
+
+            //runtime if start = 0 runtime error: bad text or out of bounds
+            
+            if(end == 0 || end > text.Length + 1)
+                end = text.Length+1;
+            if(start < 0)
+                start = Math.Max(start + text.Length + 1, 1);
+            if(end < 0)
+                end = Math.Min(end + text.Length + 1, text.Length);
+            
+            if(start == 0 || start > text.Length || start > end)
+                throw new Exception("bad text or out of bounds");
+                
+            string result = text.Remove(start - 1, (end-start)).Insert(start - 1, insert_text);
+
+            return new DreamValue(result);
+        }
+
+        [DreamProc("splicetext_char")]
+        [DreamProcParameter("Text", Type = DreamValueType.String)]
+        [DreamProcParameter("Start", Type = DreamValueType.Float, DefaultValue = 1)]
+        [DreamProcParameter("End", Type = DreamValueType.Float, DefaultValue = 0)]
+        [DreamProcParameter("Insert", Type = DreamValueType.String, DefaultValue = "")]
+        public static DreamValue NativeProc_splicetext_char(DreamObject instance, DreamObject usr, DreamProcArguments arguments)
+        {
+            arguments.GetArgument(0, "Text").TryGetValueAsString(out var text);            
+            arguments.GetArgument(1, "Start").TryGetValueAsInteger(out var start);
+            arguments.GetArgument(2, "End").TryGetValueAsInteger(out var end);             
+            arguments.GetArgument(3, "Insert").TryGetValueAsString(out var insert_text);
+
+            if(text == null) //this is for BYOND compat, and causes the function to ignore start/end if text is null or empty
+                if(insert_text == null | insert_text == "")
```

Comment:
You used `|` here instead of `||`

Just use `String.IsNullOrEmpty()` instead
```suggestion
                if(String.IsNullOrEmpty(insert_text))
```